### PR TITLE
hlx strains creates entire Service Config now

### PR DESCRIPTION
**Pull request by trieloff**
As @kptdobe reported, setting up a new Fastly configuration from scratch is a task that is better left automated, so here is a first shot at automating it:

- defines backends for GitHub and Adobe I/O Runtime
- sets `helix.vcl` (what used to be `primordialsoup.vcl`) – can be overridden
- works better when VCL files have not been created before
- `run()` waits for the configuration to complete before returning
- comes with tests for updating a new service config and an existing service config

### CODE REVIEW

File: test/mocha.opts [hunk 1]
```diff
@@ -0,0 +1 @@
+--timeout 15000
```
Comment (tripodsan) [inline test/mocha.opts]: why is this here? is this global for all tests? it's better to define it for individual tests or for suites:
see https://mochajs.org/#timeouts

File: src/strain.cmd.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ const { toBase64 } = require('request/lib/helpers');
 const strainconfig = require('./strain-config-utils');
 
 const STRAIN_FILE = path.resolve(process.cwd(), '.hlx', 'strains.yaml');
+const HELIX_VCL_FILE = path.resolve(process.cwd(), 'helix.vcl');
+const HELIX_VCL_DEFAULT_FILE = path.resolve(__dirname, 'helix.vcl');
```
Comment (tripodsan) [inline src/strain.cmd.js]: can we move this outside the normal `src` folder? maybe to `layouts` ?

File: src/helix.vcl [hunk 3]
```diff
@@ -0,0 +1,1216 @@
...
+          <div class='_1'>SERVER ERROR</div>
+          <code class='btn'>There is no action with the configured prefix.</code>
+      </div>
+      </body>
+    </html>"};
+    return(deliver);
+  }
+  if (obj.status == 953 ) {
+    set obj.http.Content-Type = "text/html";
+    synthetic {"<!doctype html>
```
Comment (tripodsan) [inline src/helix.vcl]: is there a better way to do this? maybe use some sort of `include` ?
having this large vlc is rather hard to understand and maintain. also, inline HTML is rather nasty.

File: src/strain.cmd.js [hunk 4]
```diff
@@ -278,29 +356,53 @@ class StrainCommand {
...
+      // overwrite
+      Object.assign({
+        method: 'PUT',
+        form: {
+          name,
+          content: vcl,
+        },
+      }, baseopts)).catch(() =>
+      // create new
+      Object.assign({
```
Comment (tripodsan) [inline src/strain.cmd.js]: you should probably test whit the request failed, and only create a new one, for a 404.

File: src/strain.cmd.js [hunk 5]
```diff
@@ -343,86 +461,87 @@ class StrainCommand {
...
-    return this;
+      // wait for the two initialization jobs to finish, too
+      strainjobs.push(initfp);
+      strainjobs.push(initdp);
+
+      // also wait for the strain.vcl writing to finish
+      strainjobs.push(strainp);
+      strainjobs.push(secretp);
+
+      return Promise.all(strainjobs).then(() => {
```
Comment (tripodsan) [inline src/strain.cmd.js]: btw `Promise.all()` does not process the jobs sequentially... don't know if this is important.

Comment (trieloff) [reply src/strain.cmd.js]: I know, and I'm fine with that.

File: test/mocha.opts (hunk 1 above)
Comment (trieloff) [reply test/mocha.opts]: Yes, this sets the global timeout. I got impatient after chasing timeout issue after timeout issue.

File: src/strain.cmd.js (hunk 2 above)
Comment (trieloff) [reply src/strain.cmd.js]: Will do.

File: src/helix.vcl (hunk 3 above)
Comment (trieloff) [reply src/helix.vcl]: I think I can generate the `synthetic` block from example html files and then include it. Only downside is that we will have to make additional requests to create these VCL files.

File: src/strain.cmd.js (hunk 4 above)
Comment (trieloff) [reply src/strain.cmd.js]: Will do.

**Comment by trieloff** [comment]
Let's not move the functions out into a util class until we use them in more than one place.
